allensdk/test/brain_observatory/behavior/test_write_behavior_nwb.py

```diff
@@ -168,28 +171,19 @@ def test_add_rewards(nwbfile, roundtrip, roundtripper, rewards):
                                   check_dtype=False)
 
 
-@pytest.mark.parametrize('roundtrip', [True, False])
-def test_add_behavior_only_metadata(roundtrip, roundtripper,
-                                    behavior_only_metadata_fixture):
+def test_add_behavior_only_metadata(behavior_only_metadata_fixture):
 
     metadata = behavior_only_metadata_fixture
     nwbfile = pynwb.NWBFile(
         session_description='asession',
         identifier='afile',
-        session_start_time=metadata['date_of_acquisition']
+        session_start_time=metadata.date_of_acquisition
     )
-    nwb.add_metadata(nwbfile, metadata, behavior_only=True)
-
-    if roundtrip:
-        obt = roundtripper(nwbfile, BehaviorNwbApi)
-    else:
-        obt = BehaviorNwbApi.from_nwbfile(nwbfile)
+    metadata.to_nwb(nwbfile=nwbfile)
 
-    metadata_obt = obt.get_metadata()
+    metadata_obt = BehaviorMetadata.from_nwb(nwbfile=nwbfile)
```

Comment:
This test looks to be missing a "roundtrip" component. Currently this test is only asserting that given an NWBFile object that has had data added to it that it can be loaded by `BehaviorMetadata`. What is missing is the case where you add data to an `NWBFile` object write it to an actual *.nwb file and then try to load that actual *.nwb file back in to populate `BehaviorMetadata`.

Although pynwb *should* have testing to deal with those scenarios, the coersion of NWBFile object to hdmf (h5) format is not foolproof and we have previously discovered errors, hence the existence of roundtrip tests.